data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessor.java

```diff
@@ -134,6 +135,23 @@ public static long getTimeNanos(final Instant time) {
         return currentTimeNanos;
     }
 
+    public static Instant convertObjectToInstant(Object timeObject) {
```

Comment:
This should have a unit test specifically for this.

Also, you can add this as a `InstantHelper` to the `actions` package and make it package-protected. That way it doesn't expand much.

Reply:
I kept it here because this may become useful at aggregate processor level at a later time.